keps/sig-node/20190717-seccomp-ga.md

```diff
@@ -401,9 +648,84 @@ friction.
 
 ## Alternatives
 
+### Localhost profiles
 The localhost feature currently depends on an alpha Kubelet flag. We could therefore label the
 localhostProfile source as an alpha field, and keep it's functionality in an alpha state.
 
+
+### Updating PodSecurityPolicy API
+
+#### PodSecurityPolicy API
+
+```go
+type PodSecurityPolicySpec struct {
+    ...
+    // seccomp is the strategy that will dictate allowable and default seccomp
+    // profiles for the pod.
+    // +optional
+    Seccomp *SeccompStrategyOptions
+    ...
+}
+
+type SeccompStrategyOptions struct {
+    // The default profile to set on the pod, if none is specified.
+    // The default MUST be allowed by the allowedProfiles.
+    // +optional
+    DefaultProfile *v1.SeccompProfile
+
+    // The set of profiles that may be set on the pod or containers.
+    // If unspecified, seccomp profiles are unrestricted by this policy.
+    // +optional
+    AllowedProfiles *SeccompProfileSet
+}
+
+// A set of seccomp profiles. This struct should be a plural of v1.SeccompProfile.
+// All values are optional, and an unspecified field excludes all profiles of
+// that type from the set.
+type SeccompProfileSet struct {
+    // The allowed seccomp profile types.
+    // +optional
+    Types []SeccompProfileType
+    // The allowed localhostProfiles. Values may end in '*' to include all
+    // localhostProfiles with a prefix.
+    // +optional
+    LocalhostProfiles []string
+}
+```
+
+#### PodSecurityPolicy Creation
+
+Unlike with pods, PodSecurityPolicy seccomp annotations and fields are _not_ synced.
+
+If only seccomp annotations or fields are specified, no action is necessary. The set value is used
+when applying the PodSecurityPolicy.
+
+If both seccomp annotations _and_ fields are specified, the values MUST match. This will be enforced
+in API validation.
+
+#### PodSecurityPolicy Update
+
+PodSecurityPolicy seccomp fields are mutable. On an update, the same rules are applied as for
+creation, ignoring the old values.
+
+If only seccomp annotations or fields are specified in the updated PSP, no action is necessary, and
+the specified values are used.
+
+If both seccomp annotations _and_ fields are specified in the updated PSP, the values MUST match.
+
+#### PodSecurityPolicy Enforcement
```

Comment:
This section should stay (it's a requirement, not an alternative).

Reply:
Oops, good catch

Reply:
Good one - I have just reverted that.